pkg/autoscaler/autoscaler/autoscaler_manager.go

```diff
@@ -95,45 +97,80 @@ func (am *autoScalerManager) Sync(tac *v1alpha1.TidbClusterAutoScaler) error {
 	if err := am.syncAutoScaling(tc, tac); err != nil {
 		return err
 	}
-	if err := am.syncTidbClusterReplicas(tac, tc, oldTc); err != nil {
-		return err
-	}
+
 	return am.updateAutoScaling(oldTc, tac)
 }
 
-func (am *autoScalerManager) syncAutoScaling(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler) error {
-	oldTikvReplicas := tc.Spec.TiKV.Replicas
-	if err := am.syncTiKV(tc, tac); err != nil {
-		tc.Spec.TiKV.Replicas = oldTikvReplicas
-		klog.Errorf("tac[%s/%s] tikv sync failed, continue to sync next, err:%v", tac.Namespace, tac.Name, err)
-	}
-	oldTidbReplicas := tc.Spec.TiDB.Replicas
-	if err := am.syncTiDB(tc, tac); err != nil {
-		tc.Spec.TiDB.Replicas = oldTidbReplicas
-		klog.Errorf("tac[%s/%s] tidb sync failed, continue to sync next, err:%v", tac.Namespace, tac.Name, err)
+func (am *autoScalerManager) syncExternal(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, component v1alpha1.MemberType) error {
+	switch component {
+	case v1alpha1.TiDBMemberType:
+		targetReplicas, err := query.ExternalService(tc, v1alpha1.TiDBMemberType, tac.Spec.TiDB.External.Endpoint, am.kubecli)
+		if err != nil {
+			klog.Errorf("tac[%s/%s] 's query to the external endpoint got error: %v", tac.Namespace, tac.Name, err)
+			return err
+		}
+
+		if tc.Spec.TiDB.Replicas == targetReplicas {
+			return nil
+		}
+
+		updated := tc.DeepCopy()
+		updated.Spec.TiDB.Replicas = targetReplicas
+		if _, err = am.tcControl.UpdateTidbCluster(updated, &updated.Status, &tc.Status); err != nil {
```

Comment:
With the new design, we should also create new TidbCluster for the external query, so this will be done in the following PRs?

Reply:
yes, the hust team wants us to recover the external service asap